onyo/lib/onyo.py

```diff
@@ -92,6 +92,94 @@ def clear_caches(self) -> None:
         self._gitfiles = None
         self._templates = None
 
+    def generate_commit_message(self, message: list[str] = [], cmd: str = "",
+                                keys: list[str] = [], destination: str = "") -> str:
+        """
+        If `message` is given, the function uses the first element of it to
+        generate a message subject, and the following ones for the message body.
+
+        If no `message` is given, generates a default commit message based on
+        the other parameters given, and the `files staged` in the repository.
+        The paths for the message are shortened if needed, so that the resulting
+        message subject does not exceed `MAX_LEN`.
```

Comment:
Or via config file. I will add it as a parameter for now, but I think we should also at some point near-ish think about how to use the config file, since there is also the faux serial number length, and probably others.

Reply:
I agree, and had that in mind as well. :-)